ls1intum/ArTEMiS
# Weak References to Notifications

### Checklist

### Motivation and Context

Currently a strong relation between a course and its notifications exists. This means, trying to delete a course while notifications for this course exist fails due to foreign_key constraints in the database. In my opinion, this reference should be weak, which would allow deleting courses with existing notifications.

### Description

This only affects `GroupNotification`s as they are the only kind of notification with a reference to a course. We ensure deletion of a course by deleting related `GroupNotification`s first.

### Steps for Testing

1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Create a new Course and navigate to Course Exercises
4. Create a new Quiz with a multiple choice question
5. Set the new quiz visible (which creates a new group notification for this course)
6. On Course Administration, delete the course
7. Course should be deleted and the api call `DELETE http://localhost:8080/api/courses/{courseId}` should not fail with status 500 (`could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement`)
